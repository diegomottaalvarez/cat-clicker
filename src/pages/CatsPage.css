div.main-container {
  display: flex;
  margin-left: 12rem;
  margin-right: 10rem;
}

.cat-list-container {
  width: 30vw;
  text-align: left;
}

.cat-detail-container {
  width: 70vw;
}

@media (max-width: 1080px) {
  div.main-container {
    margin-left: 5rem;
  }
}

@media (max-width: 900px) {
  div.main-container {
    flex-direction: column;
    margin-right: 0;
    margin-left: 0;
  }

  div.cat-list-container,
  div.cat-detail-container {
    width: 100%;
  }

  div.main-container.cat-selected div.cat-list-container {
    visibility: hidden;
    opacity: 0;
    height: 0;
    transition: visibility 0s, opacity 0.5s linear;
  }

  div.main-container.cat-selected.cat-detail-container.cat-selected {
    visibility: visible;
    opacity: 1;
    width: 100%;
    position: absolute;
  }
}
