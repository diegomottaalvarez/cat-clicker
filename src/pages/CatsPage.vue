<template>
  <div class="main-container">
    <CatsList v-if="catsArray" :catsList="catsArray"></CatsList>
  </div>
</template>

<script>
import CatsList from '../components/CatsList.vue';
import { getCats } from '@/helpers/getCats';

export default {
  components: { CatsList: CatsList },
  data() {
    return {
      catsArray: null,
    };
  },
  mounted() {
    this.getCatsList();
  },

  methods: {
    async getCatsList() {
      this.catsArray = await getCats();
    },
  },
};
</script>

<style>
div .main-container {
  display: flex;
  max-width: 50vw;
}
</style>
