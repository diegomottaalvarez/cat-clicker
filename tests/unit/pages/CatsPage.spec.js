import { shallowMount } from '@vue/test-utils';
import CatsPage from '@/pages/CatsPage';
import { catsListMock } from '@/helpers/getCats';

describe('CatsPage component', () => {
  let wrapper;
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () =>
        Promise.resolve({
          catsListMock,
        }),
    })
  );

  beforeEach(() => {
    wrapper = shallowMount(CatsPage);
    jest.clearAllMocks();
  });

  test('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should get the catslist', async () => {
    await wrapper.vm.getCatsList();
    expect(wrapper.vm.catsArray.length).toBeGreaterThanOrEqual(0);
  });
});
