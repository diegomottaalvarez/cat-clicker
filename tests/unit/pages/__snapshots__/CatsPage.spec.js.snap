// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatsPage component should match the snapshot 1`] = `
<div class="main-container">
  <div class="cat-list-container">
    <cats-list-stub catslist="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"></cats-list-stub>
  </div>
  <div class="cat-detail-container">
    <!--v-if-->
  </div>
</div>
`;
